src/client/src/components/chatpage.js

```diff
@@ -25,18 +25,22 @@ class ChatPage extends React.Component {
 		this.sendMessage = this.sendMessage.bind(this)
 	}
 	componentDidMount() {
-		/* istanbul ignore next */
-		socket.on(protocols.CONNECTION, () => this.setState({connected: true}))
-		/* istanbul ignore next */
-		socket.on(protocols.MESSAGE, data => this.updateMessageState(data))
-		/* istanbul ignore next */
-		socket.on(protocols.JOIN, data => this.updateJoinedRooms(data))
-		/* istanbul ignore next */
+		socket.on(protocols.CONNECTION, this.handleSocketData)
```

Comment:
Are you sure that you won't lose binding to `this` due to the fact that it's being used by `socket.on` as callback? Same question to other cases.

Reply:
Confirmed, https://pastebin.com/aDAewCXf for details